### PR TITLE
fix: make CreateBucketRequest extend from BucketMetadata to allow all…

### LINKED ISSUE #2454: CreateBucketRequest type is misaligned with JSON API
Currently the [`CreateBucketRequest` type](https://github.com/googleapis/nodejs-storage/blob/c267e6b81e8d0d19f6c7ef886c6897e4bfc7897b/src/storage.ts#L119) is missing some fields [that are supported by the JSON API ](https://cloud.google.com/storage/docs/json_api/v1/buckets/insert). 

Short term: add the missing fields
Long term: Use a single metadata / resource representation type

### PR DESCRIPTION
**Pull request by ddelgrosso1**
… fields to be set

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #2454  🦕

### CODE REVIEW

**Comment by ddelgrosso1** [comment]
Making `CreateBucketReqest` extend `BucketMetadata` so that the missing fields that are accepted by https://cloud.google.com/storage/docs/json_api/v1/buckets/insert get carried over. I've left the custom fields alone as that would be a breaking change.

File: src/storage.ts [hunk 1]
```diff
@@ -117,30 +131,30 @@ export interface AutoclassConfig {
...
+    | 'authenticatedRead'
+    | 'bucketOwnerFullControl'
+    | 'bucketOwnerRead'
+    | 'private'
+    | 'projectPrivate'
+    | 'publicRead';
+  projection?: 'full' | 'noAcl';
   regional?: boolean;
   requesterPays?: boolean;
-  retentionPolicy?: object;
```
Comment (danielduhh) [inline src/storage.ts:143]: will removing this cause any issues?

**Review by danielduhh — COMMENTED** [review]
tests?

File: src/storage.ts (hunk 1 above)
Comment (ddelgrosso1) [reply src/storage.ts:143]: It isn't removed. This interface now extends `BucketMetadata` https://github.com/googleapis/nodejs-storage/blob/18eef67892af6c7311a12d4c22579e240149eb02/src/bucket.ts#L347

**Comment by ddelgrosso1** [comment]
@danielduhh all tests are existing. Didn't make much sense to add unit tests to test each individual field is settable.

**Review by danielduhh — APPROVED** [review]

**Pull request merged**